### ISSUE TITLE
feature: connection setting to have Beekeeper prompt you when doing update/insert (for production dbs)

### ISSUE THREAD

**Issue by nicolaslabbe** [beekeeper-studio/beekeeper-studio#124]
First your product is amazing!
smooth, fast, intuitive to use.

Usually I use postico for postgresql and I think your product is much more efficient.

I can't use it on right now because of some small reason.

**I would use it if:**
- Just like postico It would be nice to have different colours between environment tab. (change the whole application in red if I am in production, allow me to change one color like the dark and light them 🙏)

**i would recommend it for:**
- Add a prompt (same as colors env) when I'm in production env I don't want to edit without a confirm popup

**I'd pay for it:**
- last but not least, event if your product is open source and free I would be interested in a shared environment of saved SQL queries. So I can be shared them with my team (like insomnia software) and would be happy to pay some price if you plan to make something of an organisation plan of beekeeper.
Just saying :)

💪

**Comment by rathboma** [comment]
Aw, thanks for the kind words :-)

Def the colors thing is on the radar. Same with some sanity around production awareness

We've definitely thought about team features. If the project continues to be popular that type of service might be possible :+)

**Comment by sundowndev** [comment]
I agree about features but I'd not pay an enterprise version, I'd donate to support the development.

**Comment by kendallroth** [comment]
I would also recommend a warning when modifying (not reading) data on an instance labeled as "production"! Color coding connections is a very good step in the direction of production awareness, but physical prompts make it that much safer to perform risky operations (or more likely, prevent them from being performed).

**Comment by rathboma** [comment]
@sundowndev I never responded, but I meant to say -- we'll never hide editor features behind an 'enterprise' version. That said we both want to work on this full time, so we're thinking about some online services to go along with the app, similar to how [insomnia](https://insomnia.rest) does it.

@kendallroth our table editing will not auto-save changes, but will require you to click `save/commit`, or `discard`, so no accidentially changing values :-). Haven't formalized the concept of a `production` connection yet, and if we're careful with the UI I'm not 100% sure we need to.

**Comment by kendallroth** [comment]
@rathboma ah, what I was meaning was something like DBeaver (I think?), where `UPDATE`, `DELETE`, etc operations will prompt the user for confirmation when running on a production connection (I believe it is an optional toggle?). So it's a step further than clicking `save/commit` or hitting `ctrl + enter`, to prompt the user to make sure they know what is going to be run (ie. selected, whatver). Does that make sense?

**Comment by rathboma** [comment]
@nicolaslabbe Hey - I have a beta test for team-sharing of queries and connections coming soon, would you be interested in being a beta tester?

**Comment by rathboma** [comment]
Tracking this in #902
